### PR TITLE
DOC: Fix DataFrame.to_xarray doctests and allow the CI to run it.

### PR DESCRIPTION
**Pull request by Moisan**
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Fix the docstring for DataFrame.to_xarray as described in #22459 . I removed the pandas.panel example since it was deprecated and I replaced it with a MultiIndex.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -2492,17 +2492,25 @@ def to_xarray(self):
         """
         Return an xarray object from the pandas object.
 
+        Return the xarray equivalent of the pandas object. `xarray
+        <http://xarray.pydata.org/en/stable/>`__ is a
+        Python package that allows to handle N-dimensional data.
+
```
Comment (gfyoung) [inline pandas/core/generic.py]: I'm not sure if we need to mention this here.

Comment (Moisan) [reply pandas/core/generic.py]: I was trying to extend the previous short summary. Do you suggest to simply remove it or is there something else we should mention here?

Comment (gfyoung) [reply pandas/core/generic.py]: I think we can remove it.  We describe `xarray` elsewhere in the docs anyhow.

**Review by gfyoung — APPROVED** [review]
cc @jreback @datapythonista @WillAyd

**Review by datapythonista — COMMENTED** [review]
Thanks for the changes @Moisan 

I added some comments about the original docstring, that if you implement them, I think the examples won't only pass the tests, but will be much clearer.

File: pandas/core/generic.py [hunk 2]
```diff
@@ -2494,80 +2494,91 @@ def to_xarray(self):
...
+        >>> index = pd.MultiIndex(levels=[[pd.to_datetime("2018-01-01"),
+        ... pd.to_datetime("2015-05-23"), pd.to_datetime("2015-06-06"),
+        ... pd.to_datetime("2011-02-13"), pd.to_datetime("2014-07-06")],
+        ...    ['one', 'two']],
+        ...   labels=[[0, 0, 1, 1, 2, 2, 3, 3], [0, 1, 0, 1, 0, 1, 0, 1]],
+        ...   names=['first', 'second'])
+
+        >>> s = pd.Series(np.arange(8), index=index)
+        >>> s
```
Comment (datapythonista) [inline pandas/core/generic.py]: I find this too complicated for what we need to show. To have a Series with a multiindex with a datetime level, we can have something like:

```
import pandas as pd

df = pd.DataFrame({'date': pd.to_datetime(['2018-01-01', '2018-01-01', '2018-01-02', '2018-01-02']),
                   'animal': ['falcon', 'parrot', 'falcon', 'parrot'],
                   'speed': [350, 18, 361, 15]}).set_index(['date', 'animal'])
df['speed']
```

I haven't used much xarray myself, and not sure what makes sense to show here. May be:
- Series.to_xarray()
- DataFrame.to_xarray()
- DataFrame(with multiindex including datetime).to_xarray()

If that makes sense, I think with the first example, we can have `df.to_xarray()` and `df['max_speed'].to_xarray()`, and then a example like the one I wrote.

@jreback does this make sense?

Sorry for requesting the changes @Moisan, but my I find like the current version gives the idea that we're trying to show something more complex than what we are actually showing.

**Review by datapythonista — COMMENTED** [review]
Looks really great. Just a couple of comments that if I'm right, should simplify the examples.

File: pandas/core/generic.py (hunk 2 above)
Comment (Moisan) [reply pandas/core/generic.py]: No problem, I'm happy to make the examples more relevant :).

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @Moisan 

@jreback I think you comments should be addressed properly, let us know if that's not the case

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**